version: '2'

volumes:
  postgres_data_dev: {}
  postgres_backup_dev: {}

services:
  postgres:
    build: ./compose/postgres
    volumes:
      - postgres_data_dev:/var/lib/postgresql/data
      - postgres_backup_dev:/backups
    environment:
      - POSTGRES_USER=network_search

  redis:
    image: redis:3.0

  celeryworker:
    build:
      context: .
      dockerfile: ./compose/django/Dockerfile-dev
    env_file: .env
    depends_on:
     - postgres
     - redis
    command: celery -A network_search.taskapp worker -l DEBUG
    volumes:
      - .:/app
    environment:
      - POSTGRES_USER=network_search
      - USE_DOCKER=yes

  django:
    build:
      context: .
      dockerfile: ./compose/django/Dockerfile-dev
    command: /start-dev.sh
    depends_on:
      - postgres
      - redis
    environment:
      - POSTGRES_USER=network_search
      - USE_DOCKER=yes
    volumes:
      - .:/app
    ports:
      - "8000:8000"
    restart: on-failure

  pycharm:
    build:
      context: .
      dockerfile: ./compose/django/Dockerfile-dev
    depends_on:
      - postgres
    environment:
      - POSTGRES_USER=network_search
    volumes:
      - .:/app



